### ISSUE TITLE
Make 2.8 the default branch and fork master with 2.78/9 tag.

### ISSUE THREAD

**Issue by asheraryam** [godotengine/godot-blender-exporter#230]
Since Blender 2.8 is officially releasing next week (with an RC released yesterday) then it might be time to fork the master branch to a new 2.79 branch and merge the 2.8 branch into master.

Maybe it can wait a bit longer, but once 2.8 is out all attention will be on it.

**Comment by Jason0214** [comment]
@akien-mga Hey Remi, do you know what's the easiest to change the master branch of a github repo? I have searched and it is saying there is some settings can be change in the repo setting. However, I have no privilege to access the settings, could you give me a help on this？ Really appreciate it.

**Comment by akien-mga** [comment]
@Jason0214 I gave you "Maintainer" access now, I don't know if that's sufficient to edit such options, you'll tell me.

For now I set the `blender2.8` branch as the default branch. That doesn't overwrite master though, it just means that `master` is no longer the default branch but just another branch.

The `master` branch should likely be branched off to `blender2.79`.

Then the `master` branch would be a bit useless. There are two options I can think of:

- Keep `master` as the default development branch, after force pushing the contents of the `blender2.8` branch on it. Once 2.80 is released, support for API changes in 2.81+ would go to the `master` branch.
- Rename `blender2.8` branch to `blender2.80` -- Blender devs are known for breaking API even in "minor" releases, so we can't be sure the API will be stable for all `2.8x`.
- Branch off the current `master` to `blender2.79` for support of the old stable version.

With this option, changes made in the `master` branch would have to be cherry-picked manually in stable branches `blender2.80` (and if wanted `blender2.79`) when relevant. Users coming to the GitHub repo would land on the `master` version (so that it's the one receiving PRs), but we could have release builds for stable versions in the Releases tab and linked in the README, so that's not too much of an issue.

The other option is:

- `blender2.8` becomes the default branch and the one receiving PRs (that's what I changed right now temporarily)
- `blender2.79` is added, branched from `master`
- `master` is reset as a copy of `blender2.8`, and from there on can also receive specific commits for the development version of Blender. Since new PRs will be made against the `blender2.8` branch, changes will have to be cherry-picked manually in the `master` branch, which can be lead to missing some changes.

I'd advocate for the first option, which is what we do for Godot releases (`master` is always the unreleased version and receives PRs, `3.1` is the stable 3.1 branch and receives cherry-picks from `master`, etc.).

Tell me what you prefer and what the branch names should be and I can do the renaming and force pushes needed.

**Comment by Jason0214** [comment]
Thank you so much for such a detailed instruction! Now I can access to the 'settings' of this repo and I believe I got all I need,  so hopefully I can do all these stuff by myself.  I also favors the first option and I will try to go through your steps when I have time, it's fairly clear with your explanation, I think I can make it right.  Thank you again!

**Comment by Jason0214** [comment]
@akien-mga Hey, Rémi

Sorry, looks like I still need some help..

I am blocked this error
```
remote: error: GH006: Protected branch update failed for refs/heads/master.
remote: error: Cannot force-push to a protected branch
```
It seems to be that I don't have privilege to **force-push** and **delete** protected branch. Also I have no access to change configuration of protected branch in my `Settings` tab.

What I am trying to do is:
1. Force push the current `blender2.8` branch to `master` and set default branch back to `master`, so that `master` becomes the active developing branch again.  This step requires a **force-push**.
2. Rename current `blender2.8` to `branch2.80`, so it becomes the stable branch for Blender 2.80. It requires a remote branch **deletion**, as rename in github is a delete and create new operation.
3. Push the current `master` branch to a new remote branch `blender2.79`, it is the stable branch for Blender2.79. I already did this.

It is not so complicated, so I will happy either you give me the privilege or you simply do it in your side. Thank you very much!

**Comment by akien-mga** [comment]
@Jason0214 I made you admin, so you should be able to de-protect the `master` and `blender2.8` branch to do the necessary changes. Make sure to readd the protection eventually to restrict force pushes by default.

Note that pending PRs might break when the branches are rebased or renamed, so they might need to be merged before, or rejected, or merged manually later on.

**Comment by Jason0214** [comment]
Thanks @akien-mga, I think now it's all solved.  I will create another issue explain what is happening in case user find their branch broken.
